curriculum/challenges/english/20-upcoming-python/learn-regular-expressions-by-building-a-password-generator/657db4cb77190e33a20e852a.md

```diff
@@ -0,0 +1,67 @@
+---
+id: 657db4cb77190e33a20e852a
+title: Step 69
+challengeType: 20
+dashedName: step-69
+---
+
+# --description--
+
+When you combine default arguments with keyword arguments, you are able to explicitly pass fewer arguments than those required by the function. The arguments that are not explicitly passed to the function call will take their default values.
```

Comment:
>The arguments that are not explicitly passed to the function call will take their default values.

Unfortunately at this point it's still possible to use positional arguments, or mix the two.

To fully not allow using positional arguments in function, signature can be changed to:
```python
def generate_password(*, length=16, nums=1, special_chars=1, uppercase=1, lowercase=1):
```
Any parameter that would be before the `*`, could be used as positional argument, after it are keyword-only.